File: backend/FinanceApp.Domain/Entities/Person.cs
using FinanceApp.Domain.Enums;

namespace FinanceApp.Domain.Entities;

public class Person : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public PersonType PersonType { get; set; }
    public string? Department { get; set; }
    public string? Position { get; set; }
    public string? IdNumber { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? BankAccount { get; set; }
    public string? BankName { get; set; }
    public DateTime? JoinDate { get; set; }
    public DateTime? LeaveDate { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public ICollection<TagBinding> TagBindings { get; set; } = new List<TagBinding>();
}

File: frontend/src/features/master-data/persons/types/person.ts
export interface Person {
  id: number
  name: string
  personType: string
  department?: string
  position?: string
  idNumber?: string
  phone?: string
  email?: string
  bankAccount?: string
  bankName?: string
  joinDate?: string
  leaveDate?: string
  isActive: boolean
  createdAt: string
}

export interface PersonStatistics {
  totalCount: number
  activeCount: number
  inactiveCount: number
  thisMonthNewCount: number
}

export interface CreatePersonRequest {
  name: string
  personType: string
  department?: string
  position?: string
  idNumber?: string
  phone?: string
  email?: string
  bankAccount?: string
  bankName?: string
  joinDate?: string
}

export interface UpdatePersonRequest {
  name: string
  personType: string
  department?: string
  position?: string
  idNumber?: string
  phone?: string
  email?: string
  bankAccount?: string
  bankName?: string
  joinDate?: string
  leaveDate?: string
  isActive: boolean
}

export interface PersonFinanceSummary {
  totalCost: number
  directCost: number
  allocatedCost: number
  transactionCount: number
  totalReceivable: number
  totalReceived: number
  receivableRemaining: number
  receivableOverdueCount: number
  receivableOverdueAmount: number
  totalPayable: number
  totalPaid: number
  payableRemaining: number
  payableOverdueCount: number
  payableOverdueAmount: number
  projectCount: number
}
